File: clients/android/app/src/main/java/com/christmedical/app/AppConfig.kt
package com.christmedical.app

/**
 * Configuration for the Christ Medical native shell.
 */
object AppConfig {
    /** Production login portal loaded by the WebView. */
    const val PORTAL_URL = "https://login.christmedical.com/"

    /**
     * Returns true when [url] is within a Christ Medical host (including subpaths).
     */
    fun isAllowedNavigation(url: String): Boolean {
        val host = runCatching { java.net.URI(url).host?.lowercase() }.getOrNull()
            ?: return false
        return host == "login.christmedical.com" || host.endsWith(".christmedical.com")
    }
}

File: mobile/ios/Sources/ChristMedicalKit/AppConfig.swift
import Foundation

/// Configuration for the Christ Medical native shell.
public enum AppConfig {
    /// Production login portal loaded by the WebView.
    public static let portalURLString = "https://login.christmedical.com/"

    /// Parsed portal URL, or `nil` if the constant is malformed.
    public static var portalURL: URL? {
        URL(string: portalURLString)
    }

    /// Returns `true` when `url` is within the Christ Medical login host (including subpaths).
    public static func isAllowedNavigation(_ url: URL) -> Bool {
        guard let host = url.host?.lowercased() else { return false }
        return host == "login.christmedical.com" || host.hasSuffix(".christmedical.com")
    }
}
